File: app/src/main/java/soup/movie/data/utils/MovieAppUtil.java
package soup.movie.data.utils;

import android.content.ActivityNotFoundException;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.support.annotation.NonNull;

public class MovieAppUtil {

    private MovieAppUtil() {}

    private static final String PACKAGE_NAME_CGV = "com.cgv.android.movieapp";

    public static void executeCgvApp(@NonNull Context ctx) {
        final String pkgName = PACKAGE_NAME_CGV;
        if (!executeApp(ctx, pkgName)) {
            executePlayStoreForApp(ctx, pkgName);
        }
    }

    private static boolean executeApp(@NonNull Context ctx, @NonNull String pkgName) {
        Intent launchIntent = ctx.getPackageManager().getLaunchIntentForPackage(pkgName);
        if (launchIntent != null) {
            ctx.startActivity(launchIntent);
            return true;
        }
        return false;
    }

    private static void executePlayStoreForApp(@NonNull Context ctx, @NonNull String pkgName) {
        try {
            ctx.startActivity(new Intent(Intent.ACTION_VIEW,
                    Uri.parse("market://details?id=" + pkgName)));
        } catch (ActivityNotFoundException e) {
            ctx.startActivity(new Intent(Intent.ACTION_VIEW,
                    Uri.parse("https://play.google.com/store/apps/details?id=" + pkgName)));
        }
    }

    private static void executeWebPage(@NonNull Context ctx, @NonNull String url) {
        Intent webIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
        ctx.startActivity(webIntent);
    }

}

File: app/src/main/java/soup/movie/util/MovieAppUtil.kt
package soup.movie.util

import android.content.ActivityNotFoundException
import android.content.Context
import android.content.Intent
import android.content.Intent.ACTION_VIEW
import android.net.Uri

object MovieAppUtil {

    private const val PACKAGE_NAME_CGV = "com.cgv.android.movieapp"

    @JvmStatic
    fun executeCgvApp(ctx: Context) {
        val pkgName = PACKAGE_NAME_CGV
        if (!executeApp(ctx, pkgName)) {
            executePlayStoreForApp(ctx, pkgName)
        }
    }

    private fun executeApp(ctx: Context, pkgName: String): Boolean {
        val launchIntent = ctx.packageManager.getLaunchIntentForPackage(pkgName)
        if (launchIntent != null) {
            ctx.startActivity(launchIntent)
            return true
        }
        return false
    }

    private fun executePlayStoreForApp(ctx: Context, pkgName: String) {
        try {
            ctx.startActivity(Intent(
                    ACTION_VIEW,
                    Uri.parse("market://details?id=$pkgName")))
        } catch (e: ActivityNotFoundException) {
            ctx.startActivity(Intent(ACTION_VIEW,
                    Uri.parse("https://play.google.com/store/apps/details?id=$pkgName")))
        }

    }

    private fun executeWebPage(ctx: Context, url: String) {
        val webIntent = Intent(ACTION_VIEW, Uri.parse(url))
        ctx.startActivity(webIntent)
    }
}
